test/hotspot/jtreg/runtime/CompressedOops/CompressedClassSpaceSize.java

```diff
@@ -56,20 +56,42 @@ public static void main(String[] args) throws Exception {
               .shouldHaveExitValue(1);
 
 
-        // Maximum size is 3GB
-        pb = ProcessTools.createLimitedTestJavaProcessBuilder("-XX:CompressedClassSpaceSize=4g",
+        // Maximum size is 4GB
+        long max_class_space_size = 4L * 1024 * 1024 * 1024;
+        pb = ProcessTools.createLimitedTestJavaProcessBuilder("-XX:CompressedClassSpaceSize=" + max_class_space_size + 1,
                                                               "-version");
         output = new OutputAnalyzer(pb.start());
         output.shouldContain("outside the allowed range")
               .shouldHaveExitValue(1);
 
+        // Without CDS, we should get 4G
+        pb = ProcessTools.createLimitedTestJavaProcessBuilder("-XX:CompressedClassSpaceSize=" + max_class_space_size,
+                "-Xshare:off", "-Xlog:metaspace*", "-version");
+        output = new OutputAnalyzer(pb.start());
+        output.shouldMatch("Compressed class space mapped at: 0x\\w+-0x\\w+, reserved size: " + max_class_space_size)
+                .shouldHaveExitValue(0);
+
+        // With CDS, class space should fill whatever the CDS archive leaves us (modulo alignment)
+        pb = ProcessTools.createLimitedTestJavaProcessBuilder("-XX:CompressedClassSpaceSize=" + max_class_space_size,
+                "-Xshare:on", "-Xlog:metaspace*", "-version");
```

Comment:
The `vm.cds` property indicates whether the JDK was built with INCLUDE_CDS. It doesn't guarantee that a CDS archive is included in the JDK -- a JDK can be configured with `--disable-cds-archive`.

It's better to generate a dummy CDS archive and use it explicitly in the test.